docs/union.md

```diff
@@ -0,0 +1,150 @@
+# Unions
+
+Consider this Oomph code:
+
+    func foo(Str | null message):
+        if message == null:
+            print("No message")
+        else:
+            print(message.get())
+
+    export func main():
+        foo(null)       # prints "No message"
+        foo("hello")    # prints "hello"
+
+Let's go through this step by step.
+
+    func foo(Str | null message):
+
+Here `Str | null` is a union type. Read it as "`Str` or `null`".
+When the code runs, each instance of the union type has exactly one **active member**,
+such as `Str` when this function is called with the string `"hello"` as an argument.
+A union type can also contain multiple types, such as `Str | Int | Bool`.
```

Comment:
Maybe I got it wrong but if I understand correctly, the 'Str | null' message means actually 'Str | Int | Bool' message? If so, you should add an emphasis to it, because upon first look it seems it only accepts a str or a null message.